### ISSUE TITLE
💅 noUnsafeOptionalChaining misses optional chain used as constructor target through TypeScript `as` assertion

### ISSUE THREAD

**Issue by earonesty** [biomejs/biome#10698]
## Description

  `lint/correctness/noUnsafeOptionalChaining` does not report an unsafe optional chain when the optional chain is used as a `new` constructor target through a TypeScript `as` assertion.

  The same unsafe pattern is reported correctly when the TypeScript assertion is removed.

  This matters because the missed form can later be emitted by a bundler/minifier as invalid JavaScript, producing:

  ```text
  SyntaxError: Invalid optional chain from new expression
  ```

  MDN reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Bad_new_optional

  ## Reproduction

  const arrays = [new Float64Array(1)];

  const out = new (arrays[0]?.constructor as { new (length: number): Float64Array })(1);

  Run:

  pnpm exec biome check repro.ts

  ## Actual Result

  Biome does not report lint/correctness/noUnsafeOptionalChaining.

  ## Expected Result

  Biome should report lint/correctness/noUnsafeOptionalChaining, because the optional chain may evaluate to undefined, and that value is being used as a constructor target.

  ## Comparison

  This similar JavaScript form is reported correctly:

  const arrays = [new Float64Array(1)];

  const out = new (arrays[0]?.constructor)(1);

  Biome reports:

  lint/correctness/noUnsafeOptionalChaining
  Unsafe usage of optional chaining.

  ## Notes

  It looks like the TypeScript as assertion may be interrupting the parent walk in noUnsafeOptionalChaining before the rule reaches the surrounding JsNewExpression.

  A possible fix would be for the rule to traverse through TypeScript assertion expressions, such as TsAsExpression, similarly to how it already traverses through parenthesized expressions.

  ## Environment

<details>

```bash

  CLI:
    Version:                      2.5.0
    Color support:                false

  Platform:
    CPU Architecture:             x86_64
    OS:                           linux

  Environment:
    BIOME_DISTRIBUTION:                npm
    BIOME_LOG_PATH:                    unset
    BIOME_LOG_PREFIX_NAME:             unset
    BIOME_LOG_LEVEL:                   unset
    BIOME_LOG_KIND:                    unset
    BIOME_CONFIG_PATH:                 unset
    BIOME_THREADS:                     unset
    BIOME_WATCHER_KIND:                unset
    BIOME_WATCHER_POLLING_INTERVAL:    unset
    NO_COLOR:                     1
    TERM:                         xterm-256color
    JS_RUNTIME_VERSION:           v24.14.1
    JS_RUNTIME_NAME:              node
    NODE_PACKAGE_MANAGER:         pnpm/10.30.3

  Biome Configuration:
    Status:                       Loaded successfully.
    Path:                         biome.json
    Formatter enabled:            true
    Linter enabled:               true
    Assist enabled:               true
    VCS enabled:                  true
    HTML full support enabled:    unset

  Linter:
    JavaScript enabled:           unset
    JSON enabled:                 unset
    CSS enabled:                  unset
    GraphQL enabled:              unset
    Recommended:                  unset
    Enabled rules:
      a11y/noAccessKey
      a11y/noAmbiguousAnchorText
      a11y/noAriaHiddenOnFocusable
      a11y/noAutofocus
      a11y/noDistractingElements
      a11y/noHeaderScope
      a11y/noInteractiveElementToNoninteractiveRole
      a11y/noLabelWithoutControl
      a11y/noNoninteractiveElementToInteractiveRole
      a11y/noNoninteractiveTabindex
      a11y/noPositiveTabindex
      a11y/noRedundantAlt
      a11y/noRedundantRoles
      a11y/noStaticElementInteractions
      a11y/noSvgWithoutTitle
      a11y/useAltText
      a11y/useAnchorContent
      a11y/useAriaActivedescendantWithTabindex
      a11y/useAriaPropsForRole
      a11y/useAriaPropsSupportedByRole
      a11y/useButtonType
      a11y/useFocusableInteractive
      a11y/useGenericFontNames
      a11y/useHeadingContent
      a11y/useHtmlLang
      a11y/useIframeTitle
      a11y/useKeyWithClickEvents
      a11y/useKeyWithMouseEvents
      a11y/useMediaCaption
      a11y/useSemanticElements
      a11y/useValidAnchor
      a11y/useValidAriaProps
      a11y/useValidAriaRole
      a11y/useValidAriaValues
      a11y/useValidAutocomplete
      a11y/useValidLang
      complexity/noArguments
      correctness/noUnsafeOptionalChaining
      suspicious/noExplicitAny
      suspicious/noNonNullAssertedOptionalChain
      suspicious/noUnsafeNegation
      suspicious/useIterableCallbackReturn
      ...
```

</details>

### Rule name

lint/correctness/noUnsafeOptionalChaining

### Playground link

https://biomejs.dev/playground/?tab=formatter&pane=Diagnostics&code=YwBvAG4AcwB0ACAAYQByAHIAYQB5AHMAIAA9ACAAWwBuAGUAdwAgAEYAbABvAGEAdAA2ADQAQQByAHIAYQB5ACgAMQApAF0AOwAKAGMAbwBuAHMAdAAgAG8AdQB0ACAAPQAgAG4AZQB3ACAAKABhAHIAcgBhAHkAcwBbADAAXQA%2FAC4AYwBvAG4AcwB0AHIAdQBjAHQAbwByACAAYQBzACAAewAgAG4AZQB3ACAAKABsAGUAbgBnAHQAaAA6ACAAbgB1AG0AYgBlAHIAKQA6ACAARgBsAG8AYQB0ADYANABBAHIAcgBhAHkAIAB9ACkAKAAxACkAOwAKAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoAG8AdQB0ACkA

### Expected result

 Biome should report `lint/correctness/noUnsafeOptionalChaining` 

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
My clanker can reproduce this
